Flaky test: chainer_tests/distributions_tests/test_utils.py::ModifiedXLogX

From #8371 [Travis](https://travis-ci.org/chainer/chainer/jobs/607597850?utm_medium=notification&utm_source=github_status)

```
______________________ TestModifiedXLogX_param_0_{dtype=float32, shape=(2, 3)}.test_backward_cpu _______________________
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46] self = <chainer.testing._bundle.TestModifiedXLogX_param_0_{dtype=float32, shape=(2, 3)} testMethod=test_backward_cpu>
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46]     def test_backward_cpu(self):
[2019-11-05 11:58:46] >       self.check_backward(self.x, self.gy)
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46] tests/chainer_tests/distributions_tests/test_utils.py:43: 
[2019-11-05 11:58:46] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2019-11-05 11:58:46] tests/chainer_tests/distributions_tests/test_utils.py:40: in check_backward
[2019-11-05 11:58:46]     x_data, y_grad, **self.backward_options)
[2019-11-05 11:58:46] ../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:944: in check_backward
[2019-11-05 11:58:46]     detect_nondifferentiable, is_immutable_params=False
[2019-11-05 11:58:46] ../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:463: in run
[2019-11-05 11:58:46]     self._run()
[2019-11-05 11:58:46] ../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:506: in _run
[2019-11-05 11:58:46]     self._compare_gradients(gx_numeric, gx_backward, directions)
[2019-11-05 11:58:46] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46] self = <chainer.gradient_check._CheckBackward object at 0x7f8018f87f98>, gx_numeric = array(0.06594739)
[2019-11-05 11:58:46] gx_backward = 0.07755014807363542
[2019-11-05 11:58:46] directions = (array([[-0.23456502, -0.76281573,  0.4189689 ],
[2019-11-05 11:58:46]        [-0.14629427, -0.31878032, -0.25403458]]),)
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46]     def _compare_gradients(self, gx_numeric, gx_backward, directions):
[2019-11-05 11:58:46]         atol = self.atol
[2019-11-05 11:58:46]         rtol = self.rtol
[2019-11-05 11:58:46]         # Compare the gradients
[2019-11-05 11:58:46]         try:
[2019-11-05 11:58:46]             testing.assert_allclose(
[2019-11-05 11:58:46]                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
[2019-11-05 11:58:46]         except AssertionError as e:
[2019-11-05 11:58:46]             eps = self.eps
[2019-11-05 11:58:46]             xs = self.xs
[2019-11-05 11:58:46]             gys = self.gys
[2019-11-05 11:58:46]             f = six.StringIO()
[2019-11-05 11:58:46]             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
[2019-11-05 11:58:46]                 eps, atol, rtol))
[2019-11-05 11:58:46]             for i, x in enumerate(xs):
[2019-11-05 11:58:46]                 f.write('inputs[{}]:\n'.format(i))
[2019-11-05 11:58:46]                 f.write('{}\n'.format(x))
[2019-11-05 11:58:46]             for i, gy in enumerate(gys):
[2019-11-05 11:58:46]                 f.write('grad_outputs[{}]:\n'.format(i))
[2019-11-05 11:58:46]                 f.write('{}\n'.format(gy))
[2019-11-05 11:58:46]             for i, d in enumerate(directions):
[2019-11-05 11:58:46]                 f.write('directions[{}]:\n'.format(i))
[2019-11-05 11:58:46]                 f.write('{}\n'.format(d))
[2019-11-05 11:58:46]             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
[2019-11-05 11:58:46]             f.write('gradients (backward): {}\n'.format(gx_backward))
[2019-11-05 11:58:46]             f.write('\n')
[2019-11-05 11:58:46]             f.write('x: numeric gradient, y: backward gradient')
[2019-11-05 11:58:46]             f.write(str(e))
[2019-11-05 11:58:46] >           raise AssertionError(f.getvalue())
[2019-11-05 11:58:46] E           AssertionError: Parameterized test failed.
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           Base test method: TestModifiedXLogX.test_backward_cpu
[2019-11-05 11:58:46] E           Test parameters:
[2019-11-05 11:58:46] E             dtype: <class 'numpy.float32'>
[2019-11-05 11:58:46] E             shape: (2, 3)
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           (caused by)
[2019-11-05 11:58:46] E           AssertionError: check_backward failed (eps=1e-05 atol=0.01 rtol=0.01)
[2019-11-05 11:58:46] E           inputs[0]:
[2019-11-05 11:58:46] E           [[0.2653617  0.97955227 0.78702766]
[2019-11-05 11:58:46] E            [0.79942065 0.8388158  0.6125239 ]]
[2019-11-05 11:58:46] E           grad_outputs[0]:
[2019-11-05 11:58:46] E           [[-0.03339154  0.66924393  1.28142   ]
[2019-11-05 11:58:46] E            [ 0.7084703  -1.9439045   1.996255  ]]
[2019-11-05 11:58:46] E           directions[0]:
[2019-11-05 11:58:46] E           [[-0.23456502 -0.76281573  0.4189689 ]
[2019-11-05 11:58:46] E            [-0.14629427 -0.31878032 -0.25403458]]
[2019-11-05 11:58:46] E           gradients (numeric):  0.06594738676257705
[2019-11-05 11:58:46] E           gradients (backward): 0.07755014807363542
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           x: numeric gradient, y: backward gradient
[2019-11-05 11:58:46] E           Not equal to tolerance rtol=0.01, atol=0.01
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           (mismatch 100.0%)
[2019-11-05 11:58:46] E            x: array(0.065947)
[2019-11-05 11:58:46] E            y: array(0.07755)
[2019-11-05 11:58:46] E           
[2019-11-05 11:58:46] E           assert_allclose failed: 
[2019-11-05 11:58:46] E             shape: () ()
[2019-11-05 11:58:46] E             dtype: float64 float64
[2019-11-05 11:58:46] E             i: (0,)
[2019-11-05 11:58:46] E             x[i]: 0.06594738676257705
[2019-11-05 11:58:46] E             y[i]: 0.07755014807363542
[2019-11-05 11:58:46] E             relative error[i]: 0.1496162367097134
[2019-11-05 11:58:46] E             absolute error[i]: 0.011602761311058363
[2019-11-05 11:58:46] E             relative tolerance * |y[i]|: 0.0007755014807363542
[2019-11-05 11:58:46] E             absolute tolerance: 0.01
[2019-11-05 11:58:46] E             total tolerance: 0.010775501480736355
[2019-11-05 11:58:46] E           x: 0.06594739
[2019-11-05 11:58:46] E           y: 0.07755015
[2019-11-05 11:58:46] 
[2019-11-05 11:58:46] ../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:536: AssertionError
```

--- a/tests/chainer_tests/distributions_tests/test_utils.py
+++ b/tests/chainer_tests/distributions_tests/test_utils.py
@@ -37,7 +37,7 @@ class TestModifiedXLogX(unittest.TestCase):
     def check_backward(self, x_data, y_grad):
         gradient_check.check_backward(
             distributions.utils._modified_xlogx,
-            x_data, y_grad, **self.backward_options)
+            x_data, y_grad, dtype=numpy.float64, **self.backward_options)
 
     def test_backward_cpu(self):
         self.check_backward(self.x, self.gy)

